homeworks/week6/lily.py

```diff
@@ -0,0 +1,31 @@
+from collections import Counter
+S = "xyyzyzyx"
+T = "xyz"
+
+def minWindow(s, t):
+    counter = Counter(t)
+    min_len = float('inf')
+    min_s = ""
+    l = 0
+    for r, r_char in enumerate(s):
+        counter[r_char] -= 1
+        while is_substring(counter):
```

Comment:
Nice and clean solution! Although, interviewer will follow-up with the following question. Can you further improve time complexity?

Using `is_substring` simplify the solution but at the same time costly.

See sample solution for O(M+N) time complexity solution.